uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Assignment.php

```diff
@@ -21,4 +21,91 @@ public static function create($params) {
 
     return $instance;
   } */
+
+
+  /**
+   * Get all cases with no end dates
+   *
+   * @param array $params
+   * @param array $excludeCaseIds
+   * @param bool $excludeDeleted
+   *
+   * @return array of case and related data keyed on case id
+   */
+  public static function getUnclosedCases($params = array(), $excludeCaseIds = array(), $excludeDeleted = TRUE, $includeClosed = FALSE) {
+    //params from ajax call.
+    $where = array($includeClosed ? '(1)' : '(ca.end_date is null)');
+    if ($caseType = CRM_Utils_Array::value('case_type', $params)) {
+      $where[] = "( civicrm_case_type.title LIKE '%$caseType%' )";
+    }
+    if ($sortName = CRM_Utils_Array::value('sort_name', $params)) {
+      $config = CRM_Core_Config::singleton();
+      $search = ($config->includeWildCardInName) ? "%$sortName%" : "$sortName%";
+      $where[] = "( sort_name LIKE '$search' )";
+    }
+    if ($cid = CRM_Utils_Array::value('contact_id', $params)) {
+      $where[] = " c.id = $cid ";
+    }
+    if (is_array($excludeCaseIds) &&
+      !CRM_Utils_System::isNull($excludeCaseIds)
+    ) {
+      $where[] = ' ( ca.id NOT IN ( ' . implode(',', $excludeCaseIds) . ' ) ) ';
+    }
+    if ($excludeDeleted) {
+      $where[] = ' ( ca.is_deleted = 0 OR ca.is_deleted IS NULL ) ';
+    }
+
+    //filter for permissioned cases.
+    $filterCases = array();
+    $doFilterCases = FALSE;
+    if (!CRM_Core_Permission::check('access all cases and activities')) {
+      $doFilterCases = TRUE;
+      $session = CRM_Core_Session::singleton();
+      $filterCases = CRM_Case_BAO_Case::getCases(FALSE, $session->get('userID'));
+    }
+    $whereClause = implode(' AND ', $where);
+
+    $limitClause = '';
+    if ($limit = CRM_Utils_Array::value('limit', $params)) {
+      $limitClause = "LIMIT 0, $limit";
+    }
+
+    $query = "
+    SELECT  c.id as contact_id,
+            c.sort_name,
+            ca.id,
+            ca.subject as case_subject,
+            civicrm_case_type.title as case_type,
+            ca.start_date as start_date,
+            ca.end_date as end_date,
+            ca.status_id
+      FROM  civicrm_case ca INNER JOIN civicrm_case_contact cc ON ca.id=cc.case_id
+ INNER JOIN  civicrm_contact c ON cc.contact_id=c.id
+ INNER JOIN  civicrm_case_type ON ca.case_type_id = civicrm_case_type.id
+     WHERE  {$whereClause}
+  ORDER BY  c.sort_name, ca.end_date
+            {$limitClause}
+";
```

Comment:
By reducing the responsibilities of method, maybe we can use the API here instead of depending on the internal cases structure? We would just have to make sure that, in doing that, the permissions will be applied correctly. That is, user X will only see her own tasks (or the tasks she has access permissions)